src/System.Private.CoreLib/src/BootstrapHelpers.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Runtime
+{
+    internal static class BootstrapHelpers
+    {
+        private static unsafe int DecodeUnsigned(byte** ppStream, byte* pStreamEnd, uint* pValue)
```

Comment:
You should be able to use src\Common\src\Internal\NativeFormat\NativeFormatReader.cs